core/src/smartcontracts/isi/triggers/set.rs

```diff
@@ -507,32 +633,52 @@ impl Set {
     ///
     /// Return `false` if [`Set`] doesn't contain the trigger with the given `id`.
     pub fn remove(&mut self, id: &TriggerId) -> bool {
-        self.original_contracts.remove(id);
-        self.ids
-            .remove(id)
-            .map(|event_type| match event_type {
-                TriggeringEventType::Data => self
-                    .data_triggers
-                    .remove(id)
-                    .map(|_| ())
-                    .expect("`Set::data_triggers` doesn't contain required id. This is a bug"),
-                TriggeringEventType::Pipeline => {
-                    self.pipeline_triggers.remove(id).map(|_| ()).expect(
-                        "`Set::pipeline_triggers` doesn't contain required id. This is a bug",
-                    )
-                }
-                TriggeringEventType::Time => self
-                    .time_triggers
-                    .remove(id)
-                    .map(|_| ())
-                    .expect("`Set::time_triggers` doesn't contain required id. This is a bug"),
-                TriggeringEventType::ExecuteTrigger => {
-                    self.by_call_triggers.remove(id).map(|_| ()).expect(
-                        "`Set::by_call_triggers` doesn't contain required id. This is a bug",
-                    )
-                }
-            })
-            .is_some()
+        let Some(event_type) = self.ids.remove(id) else {
+            return false;
+        };
+
+        let blob_hash = match event_type {
+            TriggeringEventType::Data => self
+                .data_triggers
+                .remove(id)
+                .map(|trigger| match trigger.executable {
+                    LoadedExecutable::Wasm(LoadedWasm { blob_hash, .. }) => Some(blob_hash),
+                    _ => None,
+                })
```

Comment:
I think you can move this part to a separate function to not to repeat this code for every trigger type. Will be something like:

```rust
.map(extract_blob_hash)
```